lib/ansible/module_utils/facts.py

```diff
@@ -504,15 +504,28 @@ def get_date_time_facts(self):
         self.facts['date_time']['tz_offset'] = time.strftime("%z")
 
     def is_systemd_managed(self):
-        # tools must be installed
-        if self.module.get_bin_path('systemctl'):
-
-            # this should show if systemd is the boot init system, if checking init faild to mark as systemd
-            # these mirror systemd's own sd_boot test http://www.freedesktop.org/software/systemd/man/sd_booted.html
-            for canary in ["/run/systemd/system/", "/dev/.run/systemd/", "/dev/.systemd/"]:
-                if os.path.exists(canary):
-                    return True
-        return False
+        try:
```

Comment:
this detection method fails on many systems, we moved away from this to the current one to solve those issues.